### PR TITLE
Simplify some functions

### PR DESCRIPTION
**Pull request by zhangted**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 
If it fixes a bug or resolves a feature request, be sure to link to that issue.

Reduce useless passing of variables around

## Types of changes

What types of changes does your code introduce to static food data?
_Put an `x` in the boxes that apply_

- [ x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,22 +99,22 @@ and headers (the header of the csv files ) as array of string.
...
+      => assign(fileName, dataEntries)
+        => generate(i, fileName, data)
 ```
 
 `ParseDirectoryFiles` gets a directory path from call, and reads all files in the
 directory but will only pass csv files to `csvToJson(directory, file, headers)`.
 Each csv file is passed into `csvParser()``.
 
 #### `csvToJson()` require csv-Parser modules`
-Fill `result` array with all csv entries => Total entries in csv file/10,000 entries per json file => gets number of json files to be generated => store in `numberOfFiles` => Pass file name, max entries per file, number of files, and csv data array to `assign`
```
Comment (atherdon) [inline README.md]: not sure why we don't using part of a content that we have at readme before.
I mean the more data we have, it'll be better for our team mates.
right nowi i'm talking about " Total entries in csv file/10,000 entries per json file => gets number of json files to be generated" - can you update it and add to your new line?

File: README.md [hunk 2]
```diff
@@ -99,22 +99,22 @@ and headers (the header of the csv files ) as array of string.
...
+Fill `dataEntries` with all the csv data. Pass `filename` and `dataEntries` to `assign`
 
-#### `assign( fileName, maxEntriesPerFile, numberOfFiles, dataEntries )` 
-For every file => calculate start/stop indexes (0-9999,10000-19999, 20000-29999..) based on max entries per file (10000). Includes edge case for the last `stop` index to be length of `dataEntries` - 1 , because it probably wont end on a perfect 10000 multiple. Pass csv data array, start/stop indexes, file name, and currrent file counter named `i` to `generate`
+#### `assign( fileName, dataEntries )`
+Set `maxEntriesPerFile`. Calculate how many files we need to make => Length of `dataEntries` divided by `maxEntriesPerFile` => Store in `fileCount` => For each file, we calculate the `start` and `stop` indexes we need to slice from `dataEntries`. For the last file, the `stop` will be the length of `dataEntries` - 1, because it is unlikely it will end on a perfect multiple of `maxEntriesPerFile`. Then `dataEntries` is sliced at `start` to `stop` and stored in `jsonObjects`. The current file number, `i`, the `fileName`, and `jsonObjects` is passed to `generate`.
```
Comment (atherdon) [inline README.md]: can we also add this line: "calculate start/stop indexes (0-9999,10000-19999, 20000-29999..) based on max entries per file (10000)."

File: README.md [hunk 3]
```diff
@@ -99,22 +99,22 @@ and headers (the header of the csv files ) as array of string.
...
 
-#### `generate( i, fileName, start, stop, dataEntries )`– requires writeFile from sd/generator to work.
-Creates sliced array called `data` from `dataEntries[start]` to `dataEntries[stop]`. Writes `data` to json file named `fileName+i`
```
Comment (atherdon) [inline README.md]: also like this part here: "Creates sliced array called `data` from ..."

**Comment by atherdon** [comment]
tell me if i should merge it and you'll make readme changes at next OR or wait for updates

File: README.md (hunk 3 above)
Comment (zhangted) [reply README.md]: kk ill do this now

File: README.md (hunk 1 above)
Comment (atherdon) [reply README.md]: @zhangted please update this line as well

File: README.md (hunk 3 above)
Comment (atherdon) [reply README.md]: cool, then i'm waiting

File: README.md (hunk 1 above)
Comment (zhangted) [reply README.md]: Okay

**Pull request merged**